### DISCUSSION TITLE
App Router dx issue with same filenames

### DISCUSSION THREAD

**Discussion by rvdende** [vercel/next.js#66319] [App Router]
Hi, not sure where to put this but I want to add my view on the app router design choices. 

The primary issue I have with app router is the file name page.ts in every path. You end up with a code editor with a bunch of open files with the same names. This is a developer experience issue in that it becomes tedious to jump around files with the same names and you have to double check the folder its in every time. 

A possible solution would be for instance in vscode to display the folder on each tab, but I'm not sure how feasible that would be. How do you all work switching from pages to app router?

**Comment by icyJoseph** [comment]
Hi,

The discussion about file names was taken years ago. https://github.com/vercel/next.js/discussions/37136

In VSCode it is trivial with,   "workbench.editor.labelFormat": "short", 

<img width="1002" alt="Screenshot 2024-05-29 at 12 49 22" src="https://github.com/vercel/next.js/assets/21013447/9743af07-923e-42be-833c-da5dfda586d6">

I know, this is sort of, accommodating the issue to give a way out, rather than addressing it, but that change does help a lot, and now with tools such as fzf, searching for files is rather easy, `my-app-name route-name page.tsx` and boom I am where I want. 

Every time this is brought up I reflect back on how my workflow goes with `page.tsx` and the like, and well, I never perceive it. `index.js` and family are a bit more annoying...
